packages/mgt/src/components/mgt-get/mgt-get.ts

```diff
@@ -235,74 +302,92 @@ export class MgtGet extends MgtTemplatedComponent {
 
     if (this.resource) {
       try {
-        let uri = this.resource;
-        let isDeltaLink = false;
+        let cache: CacheStore<CacheResponse>;
+        const key = `${this.version}${this.resource}`;
+        let response = null;
 
-        // if we had a response earlier with a delta link, use it instead
-        if (this.response && this.response['@odata.deltaLink']) {
-          uri = this.response['@odata.deltaLink'];
-          isDeltaLink = true;
-        } else {
-          isDeltaLink = new URL(uri, 'https://graph.microsoft.com').pathname.endsWith('delta');
+        if (responseCacheEnabled() && this.cacheEnabled) {
+          cache = CacheService.getCache<CacheResponse>(cacheSchema, responsesStore);
+          const result: CacheResponse = responseCacheEnabled() ? await cache.getValue(key) : null;
+          if (result && getResponseInvalidationTime(this.cacheInvalidationPeriod) > Date.now() - result.timeCached) {
+            response = JSON.parse(result.response);
+            this.response = response;
```

Comment:
Setting the `this.response` will cause a re-render. Consider not setting `this.response` here and letting it be set below on line 395.